### DISCUSSION TITLE
fatal error: cub/cub.cuh: No such file or directory

### DISCUSSION THREAD

**Discussion by 394781865** [apache/incubator-mxnet#16726] [Q&A] [answered]
src/operator/numpy/random/./../../tensor/./././cast_storage-inl.cuh:28:23: fatal error: cub/cub.cuh: No such file or directory compilation terminated.

I encountered the above problem when compiling mxnet。what's the problem? how can i do ?

**Comment by haojin2** [comment, marked as answer]
@394781865 how are you doing the build?
Did you checkout all the submodules?
Can you try `git submodule update --init --recursive` before you build?

**Comment by samskalicky** [comment]
@mxnet-label-bot remove [Bug]

**Comment by samskalicky** [comment]
@mxnet-label-bot add [Question]
